frontend/src/pages/modelServing/screens/projects/InferenceServiceModal/DataConnectionFolderPathField.tsx

```diff
@@ -9,19 +10,47 @@ type DataConnectionFolderPathFieldProps = {
 const DataConnectionFolderPathField: React.FC<DataConnectionFolderPathFieldProps> = ({
   folderPath,
   setFolderPath,
-}) => (
-  <FormGroup fieldId="folder-path" label="Folder path">
-    <InputGroup>
-      <InputGroupText variant="plain">/</InputGroupText>
-      <TextInput
-        aria-label="folder-path"
-        type="text"
-        value={folderPath}
-        placeholder="eg. data"
-        onChange={(value) => setFolderPath(value)}
-      />
-    </InputGroup>
-  </FormGroup>
-);
+}) => {
+  type validate = 'success' | 'warning' | 'error' | 'default';
+
+  const [validated, setValidated] = React.useState<validate>('default');
+
+  const handlePathChange = (folderPath: string) => {
+    setFolderPath(folderPath);
+    setValidated('default');
+  };
+  React.useEffect(() => {
+    const timer = setTimeout(() => {
+      if (folderPath === '') {
+        setValidated('default');
+      } else if (folderPath === '/' || folderPath.includes('//')) {
```

Comment:
If you get into an error -- I think you're going to not get out of it on `paste`... can you switch your logic around `if` this, `else` set back to default?